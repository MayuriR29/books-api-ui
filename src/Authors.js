import React, { Component } from "react";
import {API_HOST} from "./config";
class Authors extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }
  render() {
    return (
      <div>
        <h1>Authors List</h1>
        {this.state.authors.map(author => {
          return <li key={author._id}>{author.name}</li>;
        })}
      </div>
    );
  }
  async getAuthors() {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    const authors = await response.json();
    this.setState({
      authors: authors
    });
  }
  async componentDidMount() {
    await this.getAuthors();
  }
}
export default Authors;
