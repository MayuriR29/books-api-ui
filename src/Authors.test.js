import React from 'react';
import Authors from './Authors';
global.fetch = require('jest-fetch-mock');
import ShallowRenderer from "react-test-renderer/shallow";

it('title is returned when render is called first ',()=>{
    const renderer=new ShallowRenderer();
    renderer.render(<Authors/>);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});
it('test title and authorlist after component mounted',async ()=>{
    fetch.mockResponseOnce(JSON.stringify([
        {
            name:"Paulo Coelho",
            age:50,
            _id:'cascwdf23213'
        },
        {
            name:"Robin Sharma",
            age:52,
            _id:'cascwdqwwdf23213'
        }
    ]));
    const renderer=new ShallowRenderer();
    renderer.render(<Authors/>);

    const instance=renderer.getMountedInstance();
    await instance.componentDidMount();

    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
})