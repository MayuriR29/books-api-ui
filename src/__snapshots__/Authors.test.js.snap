// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test title and authorlist after component mounted 1`] = `
<div>
  <h1>
    Authors List
  </h1>
</div>
`;

exports[`title is returned when render is called first  1`] = `
<div>
  <h1>
    Authors List
  </h1>
</div>
`;
